### PR TITLE
Fix for thrown JS error

### PR DESCRIPTION
**Pull request by Reggino**
Thanks for this, this is awesome, thanks!

I did however run into an issue where `NestHydrationJS` would throw an error where null values were mixed with a nested structure. 

This PR should fix that, but i'm not 100% sure the output is the intended behaviour...

### CODE REVIEW

**Comment by Reggino** [comment]
That failing test on node 0.18 should be fixed now

File: spec/NestHydrationJS/nest.spec.js [hunk 1]
```diff
@@ -1072,4 +1072,46 @@ describe('NestHydrationJS', function () {
...
+			{
+				'id': 1,
+				'nested__id': 3
+			},
+			{
+				'id': 2,
+				'nested__id': null
+			}
+		];
+		const shapeDefinition = [
```
Comment (tb) [inline spec/NestHydrationJS/nest.spec.js]: old node tests need `var` here

File: spec/NestHydrationJS/nest.spec.js [hunk 2]
```diff
@@ -1072,4 +1072,46 @@ describe('NestHydrationJS', function () {
 			});
 		});
 	});
+
+	it('Should not break on null data with complex nesting', function () {
+		const rows = [
```
Comment (tb) [inline spec/NestHydrationJS/nest.spec.js]: old node tests need `var` here

Comment (Reggino) [inline spec/NestHydrationJS/nest.spec.js]: Thanks. 

I fixed that in my branch, but the PR isn't updated automatically. I'll do that manually.

**Comment by Reggino** [comment]
I've updated the PR in #17

**Pull request closed without merging**